### DISCUSSION TITLE
FEATURE REQUEST: Cast backed enums into string or int in blade

### DISCUSSION THREAD

**Discussion by avidianity** [laravel/framework#41181] [Ideas]
Say we have this sample model:
```php
use App\Enums\MyEnum;

class MyModel extends Model {
    protected $casts = [
        'foo' => MyEnum::class
    ];
}
```

When casting eloquent properties with enums, we have to do this everytime we try to render it, otherwise it throws error:
`{{ $model->foo->value }}`

It would be great if the renderer/view engine can recognize that it's an enum and automatically return it's string/int version so we can still render it like so:
`{{ $model->foo }}`

probably something like
```php
use BackedEnum;

if($property instanceof BackedEnum) {
   return $property->value;
}
```

**Comment by henzeb** [comment]
In the meantime maybe you can use https://github.com/henzeb/enumhancer/blob/main/docs/functions.md#value

**Comment by henzeb** [comment]
Just to let you know, the newest version of the package mentioned above now supports what you proposed. https://github.com/henzeb/enumhancer/blob/main/docs/blade.md
